cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -229,6 +233,18 @@ public ScheduleState scheduleState(ProgramId program, SchedulableProgramType pro
     }
   }
 
+  @Nullable
+  public static String removeAppVersion(String scheduleId) {
+    String[] parts = scheduleId.split(":");
+    if (parts.length != 6) {
+      return null;
+    }
+
+    List<String> stringParts = new ArrayList<>(Arrays.asList(parts));
+    stringParts.remove(3);
```

Comment:
Should check whether existing version is the default version. If it's not, should still return null. When deleting a schedule with non-default version, the existing schedule with no version should not be deleted.